android/src/main/new_arch/react/renderer/components/RNLiveMarkdownSpec/MarkdownTextInputDecoratorShadowNode.cpp

```diff
@@ -0,0 +1,127 @@
+#include <fbjni/fbjni.h>
+#include <react/fabric/JFabricUIManager.h>
+#include <react/jni/ReadableNativeMap.h>
+
+#include "MarkdownTextInputDecoratorShadowNode.h"
+
+#include <react/renderer/components/androidtextinput/AndroidTextInputState.h>
+
+#include <react/renderer/components/view/conversions.h>
+#include <react/renderer/core/ComponentDescriptor.h>
+#include <yoga/Yoga.h>
+
+namespace facebook {
+namespace react {
+
+extern const char MarkdownTextInputDecoratorViewComponentName[] =
+    "MarkdownTextInputDecoratorView";
+
+void MarkdownTextInputDecoratorShadowNode::initialize() {
+  ShadowNode::traits_.unset(ShadowNodeTraits::ForceFlattenView);
+}
+
+void MarkdownTextInputDecoratorShadowNode::createCustomContextContainer() {
+  static auto customUIManagerClass = jni::findClassStatic(
+      "com/expensify/livemarkdown/CustomFabricUIManager");
+  static auto createCustomUIManager =
+      customUIManagerClass
+          ->getStaticMethod<JFabricUIManager::javaobject(
+              JFabricUIManager::javaobject, ReadableMap::javaobject, int)>(
+              "create");
+
+  const auto markdownStyle =
+      this->getProps()->rawProps["markdownStyle"];
+  const auto currentParserId =
+      this->getProps()->rawProps["parserId"].asInt();
+
+  auto const decoratorPropsRNM =
+      ReadableNativeMap::newObjectCxxArgs(markdownStyle);
+  auto const decoratorPropsRM =
+      jni::make_local(reinterpret_cast<ReadableMap::javaobject>(
+                          decoratorPropsRNM.get()));
+
+  const JFabricUIManager::javaobject &fabricUIManager =
+      this->getContextContainer()->at<JFabricUIManager::javaobject>("FabricUIManager");
+
+  const auto customUIManager = jni::make_global(createCustomUIManager(
+      customUIManagerClass, fabricUIManager,
+      decoratorPropsRM.get(), currentParserId));
+  const ContextContainer::Shared contextContainer =
+      std::make_shared<ContextContainer const>();
+  contextContainer->insert("FabricUIManager", customUIManager);
+
+  customContextContainer_ = contextContainer;
+  previousMarkdownStyle_ = markdownStyle;
+  previousParserId_ = currentParserId;
+}
+
+void MarkdownTextInputDecoratorShadowNode::tryUpdateCustomContextContainer() {
+  const auto markdownStyle =
+      this->getProps()->rawProps["markdownStyle"];
+  const auto currentParserId =
+      this->getProps()->rawProps["parserId"].asInt();
+
+  if (currentParserId != previousParserId_ || markdownStyle != previousMarkdownStyle_) {
+    const JFabricUIManager::javaobject &customUIManager =
+        this->customContextContainer_->at<JFabricUIManager::javaobject>("FabricUIManager");
+
```

Comment:
Why do we need this?

Reply:
We don't - [1c1cc2f](https://github.com/Expensify/react-native-live-markdown/pull/566/commits/1c1cc2f941f383b31b9abb0625c8e83f5f754025)

As to why it's here - I was experimenting with updating the existing mourning manager instead of creating a new one each time we need to update. I moved away from that due to the fact that it had a potential (I think) to apply new styles/parser to already mounted decorator. I seem to have missed this when cleaning it up.